thoth/storages/graph/postgres.py

```diff
@@ -1592,6 +1592,53 @@ def get_unsolved_python_package_versions_count_all(
 
             return query.count()
 
+    # SI Analyzed Python Packages
+
+    def _construct_si_analyzed_python_package_versions_query(
+        self, session: Session, index_url: Optional[str] = None
+    ) -> Query:
+        """Construct query for packages analyzed by solver and analyzed by SI."""
+        index_url = GraphDatabase.normalize_python_index_url(index_url)
+        query = session.query(PythonPackageVersion).filter(
+            PythonPackageVersion.package_version.isnot(None),
+            PythonPackageIndex.url.isnot(None),
+            PythonPackageIndex.enabled.is_(True),
+        )
+
+        if index_url is not None:
+            query = query.filter(PythonPackageIndex.url == index_url)
+
+        # We find all rows that are same in PythonPackageVersion and SIAggregated table.
+        conditions = [PythonPackageVersion.id == SIAggregated.python_package_version_id]
+
+        query = query.filter(exists().where(and_(*conditions)))
+
+        return query
+
+    def get_si_analyzed_python_package_versions_count_all(
+        self,
+        index_url: Optional[str] = None,
+        *,
+        distinct: bool = False,
```

Comment:
Should distinct be true by default?

Reply:
we have False everywhere by default, as you want to count records in the database first, then if you want you can filter out :)